src/librustc_trans/back/write.rs

```diff
@@ -557,9 +561,15 @@ unsafe fn optimize_and_codegen(cgcx: &CodegenContext,
 
         if config.emit_obj {
             let path = output_names.with_extension(&format!("{}.o", name_extra));
-            with_codegen(tm, llmod, config.no_builtins, |cpm| {
-                write_output_file(cgcx.handler, tm, cpm, llmod, &path, llvm::ObjectFileType);
-            });
+            if !cgcx.is_like_pnacl {
+                with_codegen(tm, llmod, config.no_builtins, |cpm| {
```

Comment:
Should this logic go into pnacl itself? Basically when you request an object file to be emitted from pnacl it seems like it should naturally understand that it instead just emits bitcode.